builtin/mainmenu/dlg_mg_flags.lua

```diff
@@ -0,0 +1,243 @@
+--Minetest
+--Copyright (C) 2016 srifqi, Muhammad Rifqi Priyo Susanto
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local mg_first_time = true
+
+-- Use default if not defined
+
+-- Water level
+local water_level = 1
+
+-- General attributes
+local flag_list_g = {
+	caves		= true,
+	dungeons	= true,
+	light		= true,
+	decorations	= true
+}
+
+-- v6 mapgen
+local flag_list_6 = {
+	biomeblend	= true,
+	mudflow		= true,
+	snowbiomes	= true,
+	jungles		= true,
+	flat		= false,
+	trees		= true
+}
+
+-- v7 mapgen
+local flag_list_7 = {
+	mountains	= true,
+	ridges		= true
+}
```

Comment:
What paramat meant: `floatlands` and `caverns` are missing here
https://github.com/minetest/minetest/blob/master/minetest.conf.example#L1339